__tests__/vending-machine.spec.js

```diff
@@ -0,0 +1,251 @@
+
+const VendingMachine = require('../src/vending-machine');
+
+describe('vendingMachine', () => {
+  let test = {};
+
+  beforeEach(() => {
+
+    test.inventory = {
+      1: {
+        productId: 1,
+        productName: 'mars bar',
+        productPrice: 1.75,
+        productQuantity: 10
+      },
+      2: {
+        productId: 2,
+        productName: 'twix',
+        productPrice: 2.00,
+        productQuantity: 10
+      },
+      3: {
+        productId: 3,
+        productName: 'm&ms',
+        productPrice: 2.50,
+        productQuantity: 10
+      },
+      4: {
+        productId: 4,
+        productName: 'payday',
+        productPrice: 3.15,
+        productQuantity: 10
+      }
+    }
+
+    test.vendingMachine = new VendingMachine(test.inventory);
+  });
+
+  describe('.restock', () => {
+
+    test.itemId = 0;
+    test.newItemsAdded = 10;
+
+    beforeEach(() => {
+      test.itemId++;
```

Comment:
This makes me nervous, as it looks like the `test` object isn't being reset between tests (it isn't). Tests should be able to run in any order and still work, as the units shouldn't affect each other.